Snowflake.Data/Core/Session/SFSessionProperty.cs

```diff
@@ -260,7 +260,8 @@ internal static SFSessionProperties parseConnectionString(String connectionStrin
             if (!properties.ContainsKey(SFSessionProperty.HOST) ||
                 (0 == properties[SFSessionProperty.HOST].Length))
             {
-                string hostName = String.Format("{0}.snowflakecomputing.com", properties[SFSessionProperty.ACCOUNT]);
+                string compliantAccountName = properties[SFSessionProperty.ACCOUNT].Replace('_','-');

```

Comment:
We require a possibility to disable the replacement by the parameter in a connection string similar to JDBC to mitigate the risk of privatelink connections not working: [Reasoning for JDBC](https://community.snowflake.com/s/article/Behaviour-Change-Release-information-associated-with-Snowflake-JDBC-driver-version-3-13-25)